app/core/chat/dialogue_loop.py

```diff
@@ -0,0 +1,41 @@
+from typing import List, Dict
+
+from app.core.chat.dialogue_structs.intent import IntentEnum
+from app.core.chat.dst.dst_module import DSTModule
+from app.core.chat.dst.intent_detection import AbstractIntentDetectionModule
+from app.core.chat.nlg.nlg import NLG
+from app.core.chat.task_states import TaskState, ListMoviesState
+from app.exceptions import PROCESSING_ERROR_MESSAGE, ChatProcessingException
+
+
+class DialogueLoop:
+    def __init__(
+        self,
+        dst: DSTModule,
+        intent_detector: AbstractIntentDetectionModule,
+        nlg: NLG,
+    ):
+        self.dst: DSTModule = dst
+        self.intent_detector: AbstractIntentDetectionModule = intent_detector
+        self.nlg: NLG = nlg
+
+        self.last_intent: IntentEnum = IntentEnum.UNKNOWN
+        self.states: Dict[IntentEnum, TaskState] = {
+            IntentEnum.LIST_MOVIES: ListMoviesState(),
+        }
+
+    def answer(self, chat_history: List[str], message: str) -> str:
+        try:
+            current_intent: IntentEnum = self.intent_detector.recognize_intent(message)
+            if current_intent is IntentEnum.GIVE_SUGGESTIONS:
+                return self.states[self.last_intent].generate_suggestions(self.nlg)
+            if current_intent is IntentEnum.UNKNOWN:
+                current_intent = self.last_intent
+            # NOTE: reset the state if intent changed?
```

Comment:
Bad idea. I would rather keep it in list of `active intents`. And I would keep general state as well, as it is a good way to not ask a user multiple questions about the same e.g. date to list a movie and date of reservation. To make sure this is the same date, you ask a user to confirm it, as it is right now afaik.